backend/writer.go

```diff
@@ -1,98 +1,69 @@
 package backend
 
 import (
-	"bufio"
-	"bytes"
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
 	"math"
-	"sort"
-	"strconv"
 	"strings"
+	"time"
 
 	"github.com/prometheus/prometheus/prompb"
 	log "github.com/sirupsen/logrus"
 )
 
 type MetricWriter struct {
-	prefix         string
-	sourceOverride []string
-	pool           *Pool
-	tags           map[string]string
+	prefix      string
+	tags        map[string]string
+	sender      senders.Sender
+	metricsSent int64
+	numErrors   int64
+	errorRate   float64
 }
 
 var tagValueReplacer = strings.NewReplacer("\"", "\\\"", "*", "-")
 
-type metricPoint struct {
-	Metric    string
-	Value     float64
-	Timestamp int64
-	Source    string
-	Tags      map[string]string
-}
-
-func NewMetricWriter(host, prefix string, tags map[string]string) *MetricWriter {
+func NewMetricWriter(host string, port int, prefix string, tags map[string]string) (*MetricWriter, error) {
+	sender, err := senders.NewProxySender(&senders.ProxyConfiguration{
```

Comment:
Could we support the direct sender option as well? Could be useful for POCs etc.